Plugins/GeoModel/src/GeoModelDetectorSurfaceFactory.cpp

```diff
@@ -81,62 +96,103 @@ void Acts::GeoModelDetectorSurfaceFactory::construct(
+        ACTS_VERBOSE("Found" << subvolumes.size()
+                             << "sub volumes matching the query");
 
-      if (!matches(name, *fpv)) {
-        continue;
-      }
+        for (auto &subvol : subvolumes) {
+          // get the transform of the subvolume to the global frame
+          const Transform3 &transform =
+              fpv->getAbsoluteTransform() * subvol.transform;
 
-      // Convert using the list of converters
-      bool success = false;
-
-      for (const auto &converter : m_cfg.shapeConverters) {
-        auto converted = converter->toSensitiveSurface(*fpv);
-        if (converted.ok()) {
-          // Add the element and surface to the cache
-          cache.sensitiveSurfaces.push_back(converted.value());
-          const auto &[el, sf] = converted.value();
-          success = true;
-          ACTS_VERBOSE("successfully converted "
-                       << name << " (" << vname << " / " << recType(shape)
-                       << " / " << fpv->getLogVol()->getMaterial()->getName()
-                       << ")");
-          if (!(el && sf)) {
-            throw std::runtime_error("something is nullptr");
-          }
-          break;
+          convertSensitive(subvol.volume, transform, cache.sensitiveSurfaces);
         }
-      }
 
-      if (!success) {
-        ACTS_DEBUG(name << " (" << vname << " / " << recType(shape)
-                        << ") could not be converted by any converter");
       } else {
-        volNameStems.emplace(name.substr(0, 6));
-        materials.emplace(fpv->getLogVol()->getMaterial()->getName());
+        // convert the full phys vol
+        convertSensitive(physVol, fpv->getAbsoluteTransform(nullptr),
+                         cache.sensitiveSurfaces);
       }
     }
+
     ACTS_VERBOSE("Found " << qFPV.size()
                           << " full physical volumes matching the query.");
-
-    auto streamVec = [](const auto &v) {
-      std::stringstream ss;
-      for (const auto &el : v) {
-        ss << el << " ";
-      }
-      return ss.str();
-    };
-
-    ACTS_INFO("Converted volumes (stems): " << streamVec(volNameStems));
-    ACTS_INFO("Materials of converted volumes: " << streamVec(materials));
   }
+
   ACTS_DEBUG("Constructed "
              << cache.sensitiveSurfaces.size() << " sensitive elements and "
              << cache.passiveSurfaces.size() << " passive elements");
 }
+
+void Acts::GeoModelDetectorSurfaceFactory::convertSensitive(
+    PVConstLink geoPV, const Acts::Transform3 &transform,
+    std::vector<Acts::GeoModelSensitiveSurface> &sensitives) {
+  const GeoLogVol *logVol = geoPV->getLogVol();
+  const GeoShape *shape = logVol->getShape();
+  int shapeId = shape->typeID();
+  std::string name = logVol->getName();
+  std::shared_ptr<const Acts::IGeoShapeConverter> converter =
+      Acts::GeoShapesConverters(shapeId);
+  if (converter == nullptr) {
+    ACTS_DEBUG("converter is a nullptr");
```

Comment:
No puff? If a converter is not available, I'd expect more brimbamborium.

Reply:
fiiine

Reply:
actually it can be like that I guess, in case the shape is not found, not need to puff? just move on

Reply:
Is this something you expect to happen and move on, or does this usually indicate some underlying issue that should be fixed?

Reply:
I do not think that it indicates an underlying issue, because if we get back a nullptr it would just mean that the converter for this shape is not in the `unordered_map` (for the moment that would mean that the converter is not implemented). 
I do not know if we really want to throw exception here.
I guess it depends if we want to convert everything or also skip some shapes and continue with others that can be converted?